test/Microsoft.Azure.WebJobs.Extensions.EventHubs.Tests/EventHubScaleMonitorTests.cs

```diff
@@ -0,0 +1,485 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.EventHubs;
+using Microsoft.Azure.WebJobs.EventHubs.Listeners;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Moq;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.EventHubs.UnitTests
+{
+    public class EventHubScaleMonitorTests
+    {
+        private readonly string _functionId = "EventHubsTriggerFunction";
+        private readonly string _eventHubContainerName = "azure-webjobs-eventhub";
+        private readonly string _eventHubName = "TestEventHubName";
+        private readonly string _consumerGroup = "TestConsumerGroup";
+        private readonly string _eventHubConnectionString = "Endpoint=sb://test.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=abc123=";
+        private readonly string _storageConnectionString = "DefaultEndpointsProtocol=https;AccountName=EventHubScaleMonitorFakeTestAccount;AccountKey=ABCDEFG;EndpointSuffix=core.windows.net";
+
+        private readonly Uri _storageUri = new Uri("https://eventhubsteststorageaccount.blob.core.windows.net/");
+        private readonly EventHubScaleMonitor _scaleMonitor;
+        private readonly Mock<CloudBlobContainer> _mockBlobContainer;
+        private readonly TestLoggerProvider _loggerProvider;
+        private readonly LoggerFactory _loggerFactory;
+
+        public EventHubScaleMonitorTests()

```

Comment:
In addition to these great mock tests, I wonder if there is a way for us to also have a test against real entities? I think the E2E tests in this repo write/read from an EventHub? That way we're sure everything lines up. E.g. use the test hub/consumer group/connection string the E2E tests use, write a bunch of events, and verify the monitor returns expected results?